detectable/src/test/java/com/synopsys/integration/detectable/detectables/bazel/functional/bazel/BazelExtractorTest.java

```diff
@@ -191,4 +193,70 @@ public void testMavenInstall() throws ExecutableRunnerException, IntegrationExce
+        File bazelExe = new File("/usr/bin/bazel");
+
+        // bazel cquery --noimplicit_deps "kind(j.*import, deps(//:ProjectRunner))" --output build
+        List<String> bazelArgsGetDependenciesMavenInstall = new ArrayList<>();
+        bazelArgsGetDependenciesMavenInstall.add("cquery");
+        bazelArgsGetDependenciesMavenInstall.add("--noimplicit_deps");
+        bazelArgsGetDependenciesMavenInstall.add("kind(j.*import, deps(//:ProjectRunner))");
+        bazelArgsGetDependenciesMavenInstall.add("--output");
+        bazelArgsGetDependenciesMavenInstall.add("build");
+        ExecutableOutput bazelCmdExecutableOutputGetDependenciesMavenInstall = Mockito.mock(ExecutableOutput.class);
+        Mockito.when(bazelCmdExecutableOutputGetDependenciesMavenInstall.getReturnCode()).thenReturn(0);
+        Mockito.when(bazelCmdExecutableOutputGetDependenciesMavenInstall.getStandardOutput()).thenReturn(
+            "jvm_import(\n  name = \"com_google_guava_failureaccess\",\n" +
+                "  tags = [\"maven_coordinates=com.google.guava:failureaccess:1.0\"],\n" +
+                "  tags = [\"maven_coordinates=com.google.errorprone:error_prone_annotations:2.2.0\"],");
+        Mockito.when(executableRunner.execute(workspaceDir, bazelExe, bazelArgsGetDependenciesMavenInstall)).thenReturn(bazelCmdExecutableOutputGetDependenciesMavenInstall);
+
+        // bazel cquery --noimplicit_deps "kind(haskell_cabal_library, deps(//:ProjectRunner))" --output jsonproto
+        List<String> bazelArgsGetDependenciesHaskellCabalLibrary = new ArrayList<>();
+        bazelArgsGetDependenciesHaskellCabalLibrary.add("cquery");
+        bazelArgsGetDependenciesHaskellCabalLibrary.add("--noimplicit_deps");
+        bazelArgsGetDependenciesHaskellCabalLibrary.add("kind(haskell_cabal_library, deps(//:ProjectRunner))");
+        bazelArgsGetDependenciesHaskellCabalLibrary.add("--output");
+        bazelArgsGetDependenciesHaskellCabalLibrary.add("jsonproto");
+        ExecutableOutput bazelCmdExecutableOutputGetDependenciesHaskellCabalLibrary = Mockito.mock(ExecutableOutput.class);
+        Mockito.when(bazelCmdExecutableOutputGetDependenciesHaskellCabalLibrary.getReturnCode()).thenReturn(0);
+        File jsonProtoFile = new File("src/test/resources/detectables/functional/bazel/jsonProtoForHaskellCabalLibraries.txt");
+        String jsonProtoHaskellCabalLibrary = FileUtils.readFileToString(jsonProtoFile, StandardCharsets.UTF_8);
+        Mockito.when(bazelCmdExecutableOutputGetDependenciesHaskellCabalLibrary.getStandardOutput()).thenReturn(jsonProtoHaskellCabalLibrary);
+        Mockito.when(executableRunner.execute(workspaceDir, bazelExe, bazelArgsGetDependenciesHaskellCabalLibrary)).thenReturn(bazelCmdExecutableOutputGetDependenciesHaskellCabalLibrary);
+
+        Extraction result = bazelExtractor.extract(bazelExe, workspaceDir, bazelWorkspace, "//:ProjectRunner", new BazelProjectNameGenerator(), null, null);
```

Comment:
I think extractors should be tested in a battery test - the size of this test and the amount of mocking needed makes this test difficult to follow and it seems like a good candidate for a battery test. I wouldn't mind discussing this kind of test at large but this to me is what the battery is for.

Reply:
Moved to a new battery test